graphql_compiler/compiler/helpers.py

```diff
@@ -161,7 +161,10 @@ def validate_safe_string(value):
 
     if not set(value).issubset(VARIABLE_ALLOWED_CHARS) and \
             value not in legal_strings_with_special_chars:
-        raise GraphQLCompilationError(u'Encountered illegal characters in string: {}'.format(value))
+        raise GraphQLCompilationError(u'Encountered illegal characters in string: {}. '
+                                      u'Strings are only allowed to have underscores, '
+                                      u'digits and, upper and lower case alphabetical letters.'
+                                      .format(value))
```

Comment:
how about `legal_strings_with_special_chars`?

Reply:
Could be worth mentioning that variable in the error message. That way, if we end up changing `legal_strings_with_special_chars` later, we don't need to go find every single error message and update it or send users on a wild goose chase because of an outdated error message.

Reply:
Can't figure out a better way to express this that starting with "In general, ..."

Reply:
I think it might be time to have a separate validation function that allows `legal_strings_with_special_chars` to go through, because `validate_safe_string` is used in a lot of places now, and many of those places shouldn't allow those strings with special chars.

I'm assuming this fix is intended to improve the error message when validating output names?

Reply:
To improve the error message when validating runtime argument, tagged parameter names and output names.

Reply:
I don't think that it is worth the effort to separate validation right now. I think that the most important thing is to be as descriptive and precise as possible when raising validation errors for argument and output names since these are errors are among the most likely ones for the user to encounter.